tools/sanity_checks.py

```diff
@@ -436,16 +433,16 @@ def install_packages(kind, cmd, packages):
                 if 'unsupported' in error or 'not supported' in error or 'does not support' in error:
                     print('unsupported, as expected')
                     return
-                elif any(f"ERROR: {x}" in error for x in {'Dependency', 'Program', 'Pkg-config binary', 'CMake binary'}):
-                    if 'not found' in error:
-                        print('cannot verify in wrapdb due to missing dependency')
-                        return
-                elif 'ERROR: Could not execute Vala compiler: valac' in error:
+                if 'ERROR: Could not execute Vala compiler: valac' in error:
                     print('cannot verify in wrapdb due to missing dependency')
                     return
-                elif 'ERROR: failed to unpack archive with error: ' in error:
+                if 'ERROR: failed to unpack archive with error: ' in error:
```

Comment:
Which pylint rule says to use repeated elif? Looks funny imo...

Reply:
https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html

Reply:
Huh, weird. Looks like meson's .pylintrc disables it explicitly. :)

Reply:
Sounds like that should be undone.

Reply:
meson's pylintrc.

Reply:
What's so great about this rule? An `elif` ladder is semantically useful to visually indicate that exactly one condition (or the final "else") must be triggered. The fact that they return from the function really shouldn't matter at all.

Reply:
OTOH, less indentation is also useful.

Reply:
It's literally the same indentation though? Having an if with an elif clause versus having two if clauses one after the other, will only textually differ in eliding the `el` part of `elif`, a two-byte non-indentation change.

It's entirely unrelated to whether `else` clauses are replaced with a non-indented hunk. An if/elif must be indented since it has to check a conditional.

Reply:
it's basically the python equivalent of https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html

IMO not having the elif makes it easier to reason about. well, not in this example specifically but generally speaking.

Reply:
Pushed an alternative version. Maybe you'll like this.